plugins/telemetry-otel/build.gradle

```diff
@@ -30,6 +30,12 @@ dependencies {
   api "io.opentelemetry:opentelemetry-exporter-logging:${versions.opentelemetry}"
   api "io.opentelemetry:opentelemetry-semconv:${versions.opentelemetry}-alpha"
   api "io.opentelemetry:opentelemetry-sdk-logs:${versions.opentelemetry}"
+  api "io.opentelemetry:opentelemetry-exporter-otlp:${versions.opentelemetry}"
+  api "io.opentelemetry:opentelemetry-exporter-common:${versions.opentelemetry}"
+  api "io.opentelemetry:opentelemetry-exporter-otlp-common:${versions.opentelemetry}"
+  api "com.squareup.okhttp3:okhttp:4.10.0"
```

Comment:
Please use:

```suggestion
    runtimeOnly "io.opentelemetry:opentelemetry-exporter-sender-okhttp:${versions.opentelemetry}"
```

Reply:
Will still need to keep okhttp3 dependency.

Getting this error after removing this api "com.squareup.okhttp3:okhttp:4.10.0"

```
java.lang.IllegalStateException: SpanExporter instantiation failed for class [io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter]
	at org.opensearch.telemetry.tracing.exporter.OTelSpanExporterFactory.lambda$instantiateSpanExporter$0(OTelSpanExporterFactory.java:78) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:569) ~[?:?]
	at org.opensearch.telemetry.tracing.exporter.OTelSpanExporterFactory.instantiateSpanExporter(OTelSpanExporterFactory.java:58) ~[?:?]
	at org.opensearch.telemetry.tracing.exporter.OTelSpanExporterFactory.create(OTelSpanExporterFactory.java:49) ~[?:?]
	at org.opensearch.telemetry.tracing.OTelResourceProvider.lambda$get$0(OTelResourceProvider.java:53) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:318) ~[?:?]
	at org.opensearch.telemetry.tracing.OTelResourceProvider.get(OTelResourceProvider.java:50) ~[?:?]
	at org.opensearch.telemetry.OTelTelemetryPlugin.telemetry(OTelTelemetryPlugin.java:62) ~[?:?]
	at org.opensearch.telemetry.OTelTelemetryPlugin.getTelemetry(OTelTelemetryPlugin.java:53) ~[?:?]
	at org.opensearch.telemetry.TelemetryModule.<init>(TelemetryModule.java:28) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:596) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:400) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:404) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:180) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:171) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:104) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138) [opensearch-cli-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.cli.Command.main(Command.java:101) [opensearch-cli-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:137) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:103) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
Caused by: java.lang.ClassNotFoundException: okhttp3.Interceptor
	at java.net.URLClassLoader.findClass(URLClassLoader.java:445) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:592) ~[?:?]
	at java.net.FactoryURLClassLoader.loadClass(URLClassLoader.java:872) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	at io.opentelemetry.exporter.sender.okhttp.internal.OkHttpGrpcSenderProvider.createSender(OkHttpGrpcSenderProvider.java:42) ~[?:?]
	at io.opentelemetry.exporter.internal.grpc.GrpcExporterBuilder.build(GrpcExporterBuilder.java:158) ~[?:?]
	at io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder.build(OtlpGrpcSpanExporterBuilder.java:189) ~[?:?]
	at io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter.getDefault(OtlpGrpcSpanExporter.java:33) ~[?:?]
	at org.opensearch.telemetry.tracing.exporter.OTelSpanExporterFactory.lambda$instantiateSpanExporter$0(OTelSpanExporterFactory.java:71) ~[?:?]
```

Reply:
Removing api "com.squareup.okhttp3:okhttp:4.10.0" is causing error
```
java.lang.IllegalStateException: SpanExporter instantiation failed for class [io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter]
	at org.opensearch.telemetry.tracing.exporter.OTelSpanExporterFactory.lambda$instantiateSpanExporter$0(OTelSpanExporterFactory.java:78) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:569) ~[?:?]
	at org.opensearch.telemetry.tracing.exporter.OTelSpanExporterFactory.instantiateSpanExporter(OTelSpanExporterFactory.java:58) ~[?:?]
	at org.opensearch.telemetry.tracing.exporter.OTelSpanExporterFactory.create(OTelSpanExporterFactory.java:49) ~[?:?]
	at org.opensearch.telemetry.tracing.OTelResourceProvider.lambda$get$0(OTelResourceProvider.java:53) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:318) ~[?:?]
	at org.opensearch.telemetry.tracing.OTelResourceProvider.get(OTelResourceProvider.java:50) ~[?:?]
	at org.opensearch.telemetry.OTelTelemetryPlugin.telemetry(OTelTelemetryPlugin.java:62) ~[?:?]
	at org.opensearch.telemetry.OTelTelemetryPlugin.getTelemetry(OTelTelemetryPlugin.java:53) ~[?:?]
	at org.opensearch.telemetry.TelemetryModule.<init>(TelemetryModule.java:28) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:596) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:400) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:404) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:180) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:171) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:104) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138) [opensearch-cli-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.cli.Command.main(Command.java:101) [opensearch-cli-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:137) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:103) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
Caused by: java.lang.ClassNotFoundException: okhttp3.Interceptor
	at java.net.URLClassLoader.findClass(URLClassLoader.java:445) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:592) ~[?:?]
	at java.net.FactoryURLClassLoader.loadClass(URLClassLoader.java:872) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	at io.opentelemetry.exporter.sender.okhttp.internal.OkHttpGrpcSenderProvider.createSender(OkHttpGrpcSenderProvider.java:42) ~[?:?]
	at io.opentelemetry.exporter.internal.grpc.GrpcExporterBuilder.build(GrpcExporterBuilder.java:158) ~[?:?]
	at io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder.build(OtlpGrpcSpanExporterBuilder.java:189) ~[?:?]
	at io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter.getDefault(OtlpGrpcSpanExporter.java:33) ~[?:?]
	at org.opensearch.telemetry.tracing.exporter.OTelSpanExporterFactory.lambda$instantiateSpanExporter$0(OTelSpanExporterFactory.java:71) ~[?:?]
```

Reply:
@Shephalimittal I think we don't even need that, since the baseline is JDK-11, we could use JDK sender with no additional dependencies:

```
api "io.opentelemetry:opentelemetry-exporter-sender-jdk:${versions.opentelemetry}-alpha"
```

Reply:
I think as per https://github.com/open-telemetry/opentelemetry-java/issues/5572, We will need to add opentelemetry-exporter-sender-okhttp or opentelemetry-exporter-sender-grpc-upstream

Getting error:
```
java.lang.IllegalStateException: No GrpcSenderProvider found on classpath. Please add dependency on opentelemetry-exporter-sender-okhttp or opentelemetry-exporter-sender-grpc-upstream
[2023-09-21T01:23:13,989][ERROR][o.o.b.Bootstrap          ] [node-2] Exception
java.lang.IllegalStateException: SpanExporter instantiation failed for class [io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter]
	at org.opensearch.telemetry.tracing.exporter.OTelSpanExporterFactory.lambda$instantiateSpanExporter$0(OTelSpanExporterFactory.java:80) ~[?:?]
```

Reply:
:o this is surprise, sorry about wrong side track, it seems like we indeed have to add the dependency on okhttp, please use `runtimeOnly` scope as was suggested previously, this dependency should only appear in the plugin bundle.

Reply:
Ack.